site/_posts/2020-05-22-building-cli-app-with-device-grant-and-golang.md

```diff
@@ -0,0 +1,303 @@
+---
+layout: blog-post
+title: Building a CLI app with the Device Code grant and golang
+description: Build an authenticated CLI app using FusionAuth, the Device Code grant, and the golang client library.
+author: Matt Gowie
+image: blogs/golang-cli-device-grant/fusionauth-tutorial-building-a-cli-app-with-the-device-code-grant-and-golang.png
+category: blog
+tags: client-golang
+excerpt_separator: "<!--more-->"
+---
+
+Ever need to authenticate a user from a device that doesn't provide a great interface for logging in? Maybe your device is a smart TV, a Raspberry Pi, or perhaps a CLI app? Oh and you're a fan of Golang? Well, then this is the post for you!
+
+<!--more-->
+In this post, we'll walk through an example of how to use [FusionAuth](https://fusionauth.io) and their [Golang Client](https://github.com/FusionAuth/go-client) to authenticate a small Golang CLI app using the [OAuth2 Device Code Grant ](https://www.oauth.com/oauth2-servers/device-flow/user-flow/) ("Device Code grant").
+
+If you're a bit hazy on the Device Code grant, then here is a quick refresher from the [grant's RFC (#8628)](https://tools.ietf.org/html/rfc8628):
+
+> The OAuth 2.0 device authorization grant is designed for Internet-
+   connected devices that either lack a browser to perform a user-agent-
+   based authorization or are input constrained to the extent that
+   requiring the user to input text in order to authenticate during the
+   authorization flow is impractical.  It enables OAuth clients on such
+   devices (like smart TVs, media consoles, digital picture frames, and
+   printers) to obtain user authorization to access protected resources
+   by using a user agent on a separate device.
+
+![Device Code Grant Diagram](https://is.docs.wso2.com/en/5.11.0/assets/img/using-wso2-identity-server/deviceflow.png)
+
+## Setup Instructions
+
+First things first, to follow along you'll need golang obviously. If you don't already have golang on your local machine, *go* ahead and [install it over here](https://golang.org/doc/install).
+
+Next, we need to set up FusionAuth (FA) on our local machine to properly be able to test our CLI app. If you haven't already played around with FA then I suggest checking out their [5 minute setup guide](https://fusionauth.io/docs/v1/tech/5-minute-setup-guide) and stopping after you've completed step #3. That should get you set up with a running FA server with which our CLI app can authenticate.
+
+Once you've got FA running, we'll need to create our CLI app's FA Application. Follow these steps to do so:
+
+1. Head over to [FA's Applications UI](http://localhost:9011/admin/application/)
```

Comment:
Or to quote from our `DocsDevREADME.md` :-) 

> - If you are referencing a URL as a setting and don't want it to be hyperlinked, preface it with a `\`. For example: `\https://fusionauth-example.zendesk.com`

This is an asciidoc version, not sure exactly what the syntax is for markdown.

This is how I have been handling it... however, this may be different if you want the user to click on it while they are dev'ing locally.

Reply:
Thanks, for markdown you just don't put the []() syntax around it. TIL.